Missing SentryCrashExceptionApplication Implementation for non-MacOS Builds Causes Linker Error

### Platform

iOS

### Environment

Production

### Installed

Manually

### Version

8.41.0

### Xcode Version

16.2

### Did it work on previous versions?

8.39.0

### Steps to Reproduce

On Mac:
 
1. Clone [sentry-dotnet](https://github.com/getsentry/sentry-dotnet)
2. Change version in `modules/sentry-cocoa.properties` from `8.39.0` to `8.41.0`
3. Build the `samples/Sentry.Samples.Maui` app

### Expected Result

Build completed successfully.

### Actual Result

### Mac (Intel):

> Xamarin.Shared.Sdk.targets(1663,3): Error: clang++ exited with code 1:
> Undefined symbols for architecture x86_64:
>   “_OBJC_CLASS_$_SentryCrashExceptionApplication”, referenced from:
>     initial-undefines
> Id: symbol(s) not found for architecture x86_64
> clang++: error: linker command failed with exit code 1 (use -v to see invocation)

### Mac (M3):

> Xamarin.Shared.Sdk.targets(1663,3): Error: clang++ exited with code 1:
> Undefined symbols for architecture arm64:
>   “_OBJC_CLASS_$_SentryCrashExceptionApplication”, referenced from:
>     initial-undefines
> Id: symbol(s) not found for architecture arm64
> clang++: error: linker command failed with exit code 1 (use -v to see invocation)

### Are you willing to submit a PR?

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 
 - Use strlcpy to save session replay info path (#4740)
 - `sentryReplayUnmask` and `sentryReplayUnmask` preventing interaction (#4749)
+- missing `SentryCrashExceptionApplication` implementation for non-macOS target ()
 
 ### Improvements
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,7 +6,7 @@
 
 - Use strlcpy to save session replay info path (#4740)
 - `sentryReplayUnmask` and `sentryReplayUnmask` preventing interaction (#4749)
-- missing `SentryCrashExceptionApplication` implementation for non-macOS target ()
+- Missing `SentryCrashExceptionApplication` implementation for non-macOS target (#4759)
 
 ### Improvements
 

--- a/Sources/Sentry/Public/SentryCrashExceptionApplication.h
+++ b/Sources/Sentry/Public/SentryCrashExceptionApplication.h
@@ -9,8 +9,7 @@
 #if TARGET_OS_OSX
 #    import <AppKit/NSApplication.h>
 @interface SentryCrashExceptionApplication : NSApplication
-#else
-@interface SentryCrashExceptionApplication : NSObject
+@end
 #endif
 
-@end
+

--- a/Sources/Sentry/Public/SentryCrashExceptionApplication.h
+++ b/Sources/Sentry/Public/SentryCrashExceptionApplication.h
@@ -11,5 +11,3 @@
 @interface SentryCrashExceptionApplication : NSApplication
 @end
 #endif
-
-

